conans/test/functional/toolchains/cmake/test_cmake_toolchain.py

```diff
@@ -177,3 +177,37 @@ def build(self):
     client.run("create . pkg/0.1@")
     assert "mytoolchain1.cmake !!!running!!!" in client.out
     assert "mytoolchain2.cmake !!!running!!!" in client.out
+
+
+@pytest.mark.tool_cmake
+def test_cmaketoolchain_no_warnings():
+    """Make sure unitialized variables do not cause any warnings, passing -Werror=dev
+    and --wanr-unitialized, calling "cmake" with conan_toolchain.cmake used to fail
+    """
+    # Issue https://github.com/conan-io/conan/issues/10288
+    client = TestClient()
+    conanfile = textwrap.dedent("""
+        from conans import ConanFile
+        class Conan(ConanFile):
+            settings = "os", "compiler", "arch", "build_type"
+            generators = "CMakeToolchain", "CMakeDeps"
+            requires = "dep/0.1"
+        """)
+    consumer = textwrap.dedent("""
+       set(CMAKE_CXX_COMPILER_WORKS 1)
+       set(CMAKE_CXX_ABI_COMPILED 1)
+       project(MyHello CXX)
+       cmake_minimum_required(VERSION 3.15)
```

Comment:
The order of command calls isn't right here. `cmake_minimum_required` also sets up policies according to the provided version number. The order here results in running the toolchain with no policies. The toolchain is run via `project` while enabling a language for the first time. This should be:

```cmake
cmake_minimum_required(VERSION 3.15)
set(CMAKE_CXX_COMPILER_WORKS 1)
set(CMAKE_CXX_ABI_COMPILED 1)
project(MyHello CXX)
```

Reply:
Good to know, but not critical for this PR (it has already been merged), right? The test failed without the fixes, then passed with them. It it jsut an optimization to try to save some seconds of CI time.

Reply:
It has been merged, yes. Just letting you know that `cmake_minimum_required` should really be the very first command call in an entrypoint.